### ISSUE TITLE
[Bug]: Manual Override and Sleep Mode does not work as in OpenEVSE UI

### ISSUE THREAD

**Issue by iRonin** [firstof9/openevse#339]
### The problem

Also, `Sleep Mode (new)` does not seem to do anything.
Please check my recording:
https://www.dropbox.com/scl/fi/xunx6tg5in8262rp7n64d/OpenEVSE-sleep-override-isssue-240515.mov?rlkey=tco4ktgdv5kh8ah4fgn6arr9r&dl=0
Thanks.

### What version of Home Assistant?

core-2024.5.3

### What version of OpenEVSE?

8.2.2 & wifi v5.1.0_modified

**Comment by firstof9** [comment]
Please don't use the switches for automations, use the service calls.
The one you'll likely be using is `openevse.make_claim` in Home Assistant.

**Comment by firstof9** [comment]
The toggles are working as they should, "Sleep Mode" means charging is `disabled`.
When you change the status outside of the scheduled (auto) mode, that's an override, so that turns on.

Here's the API that's being used for reference: https://openevse.stoplight.io/docs/openevse-wifi-v4/e3153e2367cb1-open-evse-wi-fi-api

**Comment by iRonin** [comment]
make_claim is even more confusing as after making a claim from HA I cannot then use OpenEVSE UI to toggle charge…
I guess I overestimated what's possible with OpenEVSE…

After make_claim my timers are gone and UI toggle charge reverses. I guess I should release the claims or something but again I prefer how the button works...

**Comment by firstof9** [comment]
If you are charging based on solar output you should also be using the Solar PV divert, it'll adjust the available amps.

All you should really need to do is automate `active` and `disabled` via Home Assistant

**Comment by iRonin** [comment]
> If you are charging based on solar output you should also be using the Solar PV divert, it'll adjust the available amps.
> 
> All you should really need to do is automate `active` and `disabled` via Home Assistant

The Solar PV divert is not what I'm looking for. I want to make sure home batteries charging is prioritized until they are, say, 95% full while consuming solar excess during the charging and then make sue I never discharge the batteries while ,charging. The reason we're using HA is to make all the automations work like we want and not how the manufacturers want us.

**Comment by iRonin** [comment]
> The toggles are working as they should, "Sleep Mode" means charging is `disabled`. When you change the status outside of the scheduled (auto) mode, that's an override, so that turns on.
> 
> Here's the API that's being used for reference: https://openevse.stoplight.io/docs/openevse-wifi-v4/e3153e2367cb1-open-evse-wi-fi-api

I disagree because the override and sleep are somehow connected and as you can see on my video I cannot achieve the same state in the UI with integration switches.

Anyway, I think I have my logic done and will test it the following days.

Thank you for your work on this integration - it's always easier even though sometimes not everything is clear.

Perhaps mentioning integration services (openevse.make_claim etc) and providing link to OpenEVSE API in the README would save your time for issues like mine :)

**Comment by firstof9** [comment]
Yes I haven't had much time to update the README just yet, it's on my to do list.

**Comment by vermut** [comment]
I noticed the thread and I think I'm having similar issue so maybe you can point me to a right solution.

I'm trying to switch between solar_divert and just max_charge. I can do PV->Full turning off `PV`, but turning `PV` on also turns on both `sleep` buttons effectively stopping the charge.

What is the right path here?

P.S. What are `claims`?

**Comment by firstof9** [comment]
Here's the claims API: https://openevse.stoplight.io/docs/openevse-wifi-v4/221999f7bb3e5-list-evse-claims

Sleep only turns on when the EVSE is either commanded to sleep (via the toggle), or if it's set to an auto schedule and the schedule has it set to disable charging at a specific time.

**Comment by vermut** [comment]
When I touch `divert` both `sleep` follow that with 1s delay
![image](https://github.com/user-attachments/assets/9b1bd4e6-809a-4025-a73b-10e155befc74)

**Comment by firstof9** [comment]
What are your PV divert settings set to in OpenEVSE?

**Comment by firstof9** [comment]
So when you enable PV divert, that system's in control of sleeping the charger based on your settings there.

**Comment by Digital-MoBa** [comment]
If you toggle "Solar PV divert" switc,h it disable the self production. In the Web UI you can see that the ECO (Solar Symbol) is gone. This is what to expect when changing the configuration setting of self production.

To change ECO-Mode (disable/enable) you need to take the selection "openevse_divert_mode" and change fast to eco. I just need to remark that this could be done better by using a switch.

I think the most problem ist that the used update intervall for the integration is very so. That's why it's not possible to use the switch at all. For example if I flip the "Manual Override" from ON to OFF it toggle on the web-UI between AUTO-Mode and Disable Charge and the switch always stay in the ON-Position. 

It would be great if this switches work like in the web-UI and also uses the same names.

**Comment by firstof9** [comment]
> also the update interval would be much faster so the reaction of changing is noticeable

There is no update interval, the OpenEVSE pushes changes to HA.

**Comment by Digital-MoBa** [comment]
I just checked again the messages with MQTT-Explorer on the mosquitto broker that runs on my HA. There I can see all data in realtime. But the entities in the integration only change there values every 4-5 minutes. From where did the entities get there values - they are connect with the integration to the openevse IP and capure the data via tha API.

**Comment by firstof9** [comment]
> Did the openevse API do push?

Yes it pushes via websocket.

**Comment by simonjowett** [comment]
I'm in a similar position trying to relate the operation of the OpenEVSE web page with the various switches and select function in the Home assistant Integration.  I can also report the delay in updating but I think this may be the HA scan rate?  I agree with the comments above about 'eco' and 'fast', it would be easier with a switch (unless you know of future modes)!  At present I think most people either want a fast charge or a managed one.

Sleep mode does not work (as previously commented) - very confusing.

Despite the criticisms, they are not meant negatively, I very much appreciate the work - please continue!

**Comment by iRonin** [comment]
@simonjowett I've been using this integration for quite some time and everything works but you have to use the claims.

**Comment by firstof9** [comment]
> HA scan rate?

There is no scan rate, states are pushed from the OpenEVSE over a websocket connection. It's real time.

> Sleep mode does not work (as previously commented) - very confusing.

The sleep mode is mostly for backwards compatibility with older OpenEVSEs, please use the claims service call to manage sleep mode more effectively.

> I agree with the comments above about 'eco' and 'fast', it would be easier with a switch (unless you know of future modes)!

These are the modes as set in the API. The OpenEVSE team may add additional modes, I'm not sure.
What would everyone suggest a switch be named for this?

**Comment by Digital-MoBa** [comment]
I suggest that it should work like in the WebUI. Just leave the dropdown box and put all Mode inside: FAST, ECO, SLEEP. Leave a Switches for ECO-Mode to enable Solar divert and one to activate the Shaper. Write some IF statements to disable the divert and shaper switch when not ECO-Mode is selected in the dropdown. Remove all the rest switches, also the backwards compatibility ones, since they are confusing.

**Comment by vermut** [comment]
Yes, suggestion above made me understand, why this appeared like a bug to me. Indeed, webUI just have a button
![image](https://github.com/user-attachments/assets/bb41a696-101a-489f-be42-de839cabae7d)

and I was instinctively looking for the same button in HA.

**Comment by firstof9** [comment]
> Indeed, webUI just has a button

The API calls don't have an option for "auto", it would make things a bit easier.

**Comment by KipK** [comment]
In fact Auto means there's no manual override ( ro claim if you use claims ) state property. Toggling Auto on HA will just need to clear this property.

**Comment by firstof9** [comment]
> Toggling Auto on HA will just need to clear this property.

Which is what toggling "Manual override" or "Sleep mode" does.

**Comment by KipK** [comment]
I'm trying to mimic the openevse UI 3 button selectors in home assistant. 
It's not possible to do it for now we would need a way to know if there's an override and what state the override has. 
Using current manual_override sleep_mode entity won't solve the problem of displaying correct icon colors for enabled and disabled.

Something that would solve it, would be to have a select entity with the 3 states:
- Enabled ( override with state property "enabled" )
- Auto ( no manual override state )
- Disabled ( override wuith state property "disabled )

**Comment by firstof9** [comment]
> I'm trying to mimic the openevse UI 3 button selectors in home assistant.
> It's not possible to do it easily for now we would need a way to know if there's an override and what state the override has.

Have you tried the action: `list_overrides` ?

**Comment by KipK** [comment]
Yes that's what I've done, see my edited topic above. 

But it needs a template + an automation to achieve this. Not really UX friendly for some users. :)

**Comment by firstof9** [comment]
@KipK I've converted your automation there into a sensor in #428

**Comment by KipK** [comment]
> [@KipK](https://github.com/KipK) I've converted your automation there into a sensor in [#428](https://github.com/firstof9/openevse/pull/428)

Thanks, I've replaced my automation with it. Works neat.

**Comment by simonjowett** [comment]
@KipK Can you post the rest of your yaml for the Home Assistant card above, I'd like to check out this change also and replicate the OpenEVSE webpage in the H.A front end

**Comment by KipK** [comment]
> Hi, [@KipK](https://github.com/KipK) Can you please post the rest of your yaml for the Home Assistant card above, I'd like to check out this change also and replicate the OpenEVSE webpage in the H.A front end

Here is the full yaml for it: 

https://pastebin.com/G8qzEdKZ

**Comment by simonjowett** [comment]
![Image](https://github.com/user-attachments/assets/b140d91b-d21d-456d-a504-c27a17c80034)

Thanks @KipK the icons are ok - also downloaded "hui-element", but the layout is not vertical, its a grid.  I've tried different layouts ie masonary, grid etc but it does not look like your picture above.  Any advice?

**Comment by simonjowett** [comment]
Nerly there, I managed to get it very close by changing the top of the yaml to ythis:

'''
type: grid
columns: 1
square: false
cards:
'''

![Image](https://github.com/user-attachments/assets/3aff6a6d-7920-44bb-a99d-0e0b96aa110b)

**Comment by KipK** [comment]
Yes the blue part is the main HA background .

I have 3 columns on the page, EVSE is first column.

**Comment by KipK** [comment]
With latest integration and library commits/fixes this issue can be closed.

@firstof9 do you plan to bump a new version on hacs soon ?
